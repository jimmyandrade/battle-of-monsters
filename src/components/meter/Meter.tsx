import type { FC, MeterHTMLAttributes } from 'react'
import { MeterRoot } from './Meter.styles'

export const Meter: FC<MeterHTMLAttributes<HTMLMeterElement>> = (
  {
    value,
    ...props
  }
) => (
  <MeterRoot value={value} {...props} />
)
