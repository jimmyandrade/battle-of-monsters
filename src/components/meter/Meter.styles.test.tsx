import { MeterRoot } from './Meter.styles'
import { render } from '@testing-library/react'

describe('Meter styled', () => {
  describe('Root component', () => {
    it('should match snapshot', () => {
      const { container } = render(<MeterRoot />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
