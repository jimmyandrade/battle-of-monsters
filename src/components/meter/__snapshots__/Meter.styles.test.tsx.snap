// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meter styled Root component should match snapshot 1`] = `
.c0 {
  min-width: var(--meter-base-width);
  width: 100%;
}

.c0,
.c0::-webkit-meter-bar {
  background-color: var(--meter-background-color);
  border-radius: var(--meter-border-radius);
  border-width: 0;
  height: var(--meter-base-height);
  max-width: var(--meter-base-width);
  min-height: var(--meter-base-height);
}

.c0::-webkit-meter-optimum-value {
  background-color: #00FF00;
}

@media (prefers-reduced-transparency:no-preference) {
  .c0::-webkit-meter-bar,
  .c0::-webkit-meter-optimum-value {
    -webkit-transition-timing-function: ease-in-out;
    -webkit-transition-duration: var(--micro-animation-duration);
    -webkit-transition-property: width;
  }
}

<meter
  class="c0"
/>
`;
